exonum-java-binding/core/rust/src/storage/blockchain.rs

```diff
@@ -28,17 +28,11 @@ pub extern "system" fn Java_com_exonum_binding_core_blockchain_BlockchainProofs_
 ) -> jbyteArray {
     let res = panic::catch_unwind(|| {
         let name = convert_to_string(&env, full_index_name)?;
-        let db = handle::cast_handle::<View>(snapshot_handle);
-        match db.get() {
-            ViewRef::Snapshot(snapshot) => {
-                let proof = snapshot.proof_for_index(&name);
-                if let Some(proof) = proof {
-                    proto_to_java_bytes(&env, &proof)
-                } else {
-                    Ok(ptr::null_mut() as jbyteArray)
-                }
-            }
-            ViewRef::Fork(_) => panic!("nativeCreateIndexProof called with Fork"),
+        let access = handle::cast_handle::<ErasedAccess>(snapshot_handle);
+        let proof = access.proof_for_index(&name);
```

Comment:
Can we control the actual type (whether the access is Snapshot-based? Or is it controlled in ErasedAccess#proof_for_index?

Reply:
It is controlled in `ErasedAccess::proof_for_index`, it will panic and we will convert it into exception.